.github/workflows/tests.yml

```diff
@@ -41,16 +44,20 @@ jobs:
             - name: List php modules using "no php ini" mode
               run: php -m -n
             - name: Cache module
-              uses: actions/cache@v2
+              uses: actions/cache@v3
               with:
                   path: ~/.composer/cache/
                   key: composer-cache
             - name: Install dependencies
               run: composer install --no-interaction
-            - name: Run php tests
+            - name: Run shell-based test suite
+              if: runner.os == 'Linux'
               run: ./tests/launch.sh
             - name: Send coverage
-              uses: codecov/codecov-action@v1
+              if: runner.os == 'Linux'
+              uses: codecov/codecov-action@v3
               with:
                 flags: php-${{ matrix.php-version }}-${{ matrix.os }}
                 name: php-${{ matrix.php-version }}-${{ matrix.os }}
+            - name: Run PHP-based test suite
+              run: php ./tests/launch.php
```

Comment:
You can extract `./tests/launch.sh` into a matrix variable like `php-version` so line `run: ./tests/launch.sh`
is `run: ${{ matrix.test-script }}`
That will allow you so add a bunch of `- { php-version: 'x.y', experimental: false, os: ubuntu-latest, coverage-extension: 'xdebug', test-script: './tests/launch.php' }`

And this will allow you to drop your old PHP compat code. It's okay if your test suite does not coverall old php versions